src/HydraAuctionOnchain/Types/AuctionTerms.hs

```diff
@@ -48,17 +58,78 @@ newtype PAuctionTerms (s :: S)
 instance DerivePlutusType PAuctionTerms where
   type DPTStrat _ = PlutusTypeData
 
+instance PTryFrom PData PAuctionTerms
+
 data AuctionTerms
 
 type instance PlyArgOf PAuctionTerms = AuctionTerms
 
+pvalidateAuctionTerms :: Term s (PAuctionTerms :--> PUnit)
+pvalidateAuctionTerms = phoistAcyclic $
+  plam $ \auctionTerms -> P.do
+    rec <- pletAll auctionTerms
+
+    -- Auction lot must not include any ADA.
+    passert $(errCode AuctionTerms'Error'AuctionLotNonZeroAda) $
+      padaOf # rec.auctionLot #== 0
+
+    -- All amounts in the auction lot must be positive.
+    passert $(errCode AuctionTerms'Error'NonPositiveAuctionLotValue) $
+      pvaluePositive # rec.auctionLot
+
+    -- The payment part of the seller address must be seller pkh.
+    passert $(errCode AuctionTerms'Error'SellerAddressLacksPubKeyCredential) $
+      pisJust #$ paddrPaymentKeyHash # rec.sellerAddress
+
+    -- TODO: The seller pubkey hash corresponds to the seller verification key.
+    -- Note: this check only becomes possible on-chain in Plutus V3.
+    -- https://github.com/input-output-hk/plutus/pull/5431
+
+    -- Bidding ends after it the bidding start time.
+    passert $(errCode AuctionTerms'Error'BiddingStartNotBeforeBiddingEnd) $
+      pfromData rec.biddingStart #< pfromData rec.biddingEnd
+
+    -- The purchase deadline occurs after bidding ends.
+    passert $(errCode AuctionTerms'Error'BiddingEndNotBeforePurchaseDeadline) $
+      pfromData rec.biddingEnd #< pfromData rec.purchaseDeadline
+
+    -- Cleanup happens after the purchase deadline,
+    -- so that the seller can claim the winning bidder's deposit
+    -- if the auction lot is not sold.
+    passert $(errCode AuctionTerms'Error'PurchaseDeadlineNotBeforeCleanup) $
+      pfromData rec.purchaseDeadline #< pfromData rec.cleanup
+
+    -- New bids must be larger than the standing bid.
+    passert $(errCode AuctionTerms'Error'NonPositiveMinBidIncrement) $
+      0 #< pfromData rec.minBidIncrement
+
+    -- The auction fee for each delegate must contain
+    -- the min 2 ADA for the utxos that will be sent to the delegates
+    -- during fee distribution.
```

Comment:
I don't get this description, sorry. I understand the check as "minBid should cover auction fees" or something along those lines. Can you elaborate please?

Reply:
`startingBid > numDelegates * auctionFeePerDelegate > numDelegates * 2 ADA`
But I agree that the comment is a bit off.

Reply:
Addressed in https://github.com/mlabs-haskell/hydra-auction-onchain/pull/4/commits/0ebda1e5d80a41ef8e91f5713c4fb8539c749abd